internal/monitoring/interfaces.go

```diff
@@ -3,6 +3,7 @@ package monitoring
 type MonitorInterface interface {
 	GetService() string
 	GetResponseTimeMetric(map[string]string) (MetricInterface, error)
+	SetAvailability(string, float64) error
```

Comment:
given gauges and histograms implement different methods best thing is to wrap each real metric into a struct, and if type is gauge observe method would panic or do nothing, if instead it s an histogram then the set method would behave like that